### ISSUE TITLE
[Bug]: Border Token of 0 should not show any border

### ISSUE THREAD

**Issue by B3nz01d** [Orange-OpenSource/ouds-flutter#328]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-flutter/issues) for duplicate or closed feature requests

### Your test device

Any

### OS Version

Any

### App version

beta build 361

### Describe the issue

When the border is set to 0 it does show a border of 1.

### Expected Behavior

When the border is set to 0 it should not show any border.
When border value is 0, do not display any border in app tokens screen and in OUDS Flutter components.

As an example on Android, to factorize this behavior, we created a Dp extension:
```kotlin
/**
 * Returns [this] if it is not equal to [Dp.Hairline], otherwise returns `null`.
 */
val Dp.takeUnlessHairline: Dp?
    get() = takeUnless { it == Dp.Hairline }
```

It is used in tokens screen and in each component using a border token.

**Comment by AhmedAmineZr** [comment]
@B3nz01d  i made verification and I see that we use the good token, in android also we have the some behavior

**Comment by B3nz01d** [comment]
but a border of a value "0" should not appear at all, so this issue should be resolved
cc: @paulinea

**Comment by paulinea** [comment]
We have the same behavior on Android, a "0" border is considered as an "hairline" border as you can see in Android code:
```kotlin
                val hasValidBorderParams = width.toPx() >= 0f && size.minDimension > 0f
                if (!hasValidBorderParams) {
                    drawContentWithoutBorder()
                } else {
                    val strokeWidthPx =
                        min(
                            if (width == Dp.Hairline) 1f else ceil(width.toPx()),
                            ceil(size.minDimension / 2)
                        )
// ...
```
and 
```kotlin
@Stable val Hairline = Dp(0f)
```

If a user don't need any border, he just avoid using a `border` modifier.

**Comment by B3nz01d** [comment]
but from a design perspective, and in the context of the them change there could be a component that has a value of borderWidthDefault in one theme and a border of borderWidthNone in another and in that case the expected result is not a hairline, but no border at all. The code we provide should be able to handle this for the dev.

cc: @MaximeTonnerre

**Comment by MaximeTonnerre** [comment]
@B3nz01d @paulinea @AhmedAmineZr On the design side, if we use a value of 0 for a border width, the rendering is the complete visual removal of the border.

**Comment by paulinea** [comment]
@B3nz01d @MaximeTonnerre What I mean is when you use a 0 value token in a border, Android considers it as an hairline border and this is something we cannot change because this is Android behavior. If a developer don't want any border he just don't use any border modifier. But a 0 value border will always appear as an hairline on Android.
The only thing we can manage is in components, when we have a 0 value border token on something, avoid using a border modifier.

**Comment by B3nz01d** [comment]
if you can make sure that this work at least on the items we manage with some kind of if/then/else, it would be good. this could be applied to our components, but if this can be share so that a dev could use this on his/her own for the code they build, it could also be useful.

**Comment by B3nz01d** [comment]
- [ ] [Android #845](https://github.com/Orange-OpenSource/ouds-android/issues/845)

**Comment by paulinea** [comment]
@AhmedAmineZr @nouha06 @Tayebsed93 I updated this issue to explain how we finally fixed this point on Android side.

**Comment by nouha06** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (Alpha) (495)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 472
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)
cc @pya35 (accessibility team)

**Comment by nouha06** [comment]
New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (495)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 472
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)
cc @pya35 (accessibility team)

**Comment by B3nz01d** [comment]
Closed during sprint # 24
